src/components/manager/TableTreeSelect.js

```diff
@@ -1,93 +1,112 @@
-import React, {Component} from "react";
-import {connect} from "react-redux";
+import React, { useState } from "react";
+import { connect } from "react-redux";
 import PropTypes from "prop-types";
 
-import {Spin, Tag, TreeSelect} from "antd";
-
-import {nop} from "../../utils/misc";
-
-class TableTreeSelect extends Component {
-    static getDerivedStateFromProps(nextProps) {
-        if ("value" in nextProps) {
-            return {selected: nextProps.value || undefined};
-        }
-        return null;
-    }
-
-    constructor(props) {
-        super(props);
-        this.state = {selected: props.value || undefined};
-    }
-
-    onChange(selected) {
+import { Spin, Tag, TreeSelect } from "antd";
+
+const TableTreeSelect = ({
+    style,
+    value,
+    dataType,
+    onChange,
+    projects,
+    projectTables,
+    tablesByServiceID,
+    servicesLoading,
+    projectsLoading,
+}) => {
+    const [selected, setSelected] = useState(value || undefined);
+
+    const onChangeLocal = (selectedNew) => {
         // Set the state directly unless value is bound
-        if (!("value" in this.props)) this.setState({selected});
+        setSelected(selectedNew);
```

Comment:
Logic is correct, I am not quite sure the exact details of it, but I think the prop change didnt reinitialise the state, so had to change the logic to make it work. Also the state handling logic is jumbled for no reason at all. This could be made better in the future. I am gonna remove the comment for now.

Reply:
--